ui/screens.py

```diff
@@ -953,6 +999,600 @@ def walletCreationComplete(self, wallet):
+        self.layout.addWidget(lbl)
+
+        # Display info for a randomly chosen pool (with some filtering), and a
+        # couple of links to aid in selecting a VSP..
+        self.poolUrl = Q.makeLabel("", 16, a=l, fontFamily="Roboto-Medium")
+        self.poolUrl.setOpenExternalLinks(True)
+        self.poolUrl.setTextInteractionFlags(QtCore.Qt.TextSelectableByMouse);
+        scoreLbl = Q.makeLabel("score:", 14)
+        self.score = Q.makeLabel("", 14)
+        feeLbl = Q.makeLabel("fee:", 14)
+        self.fee = Q.makeLabel("", 14)
+        usersLbl = Q.makeLabel("users:", 14)
+        self.users = Q.makeLabel("", 14)
+        stats, _ = Q.makeSeries( Q.HORIZONTAL,
+            scoreLbl, self.score, Q.STRETCH,
+            feeLbl, self.fee, Q.STRETCH,
+            usersLbl, self.users
+        )
+        poolWgt, lyt = Q.makeSeries(Q.VERTICAL, self.poolUrl, stats)
+        lyt.setContentsMargins(10, 10, 10, 10)
+        lyt.setSpacing(10)
+        Q.addDropShadow(poolWgt)
+        Q.addHoverColor(poolWgt, "#f5ffff")
+        poolWgt.setCursor(QtGui.QCursor(QtCore.Qt.PointingHandCursor))
+        poolWgt.mouseReleaseEvent = self.poolClicked
+        self.layout.addWidget(poolWgt)
+
+        # A button to select a different pool and a link to the master list on
+        # decred.org.
+        btn1 = app.getButton(TINY, "show another")
+        btn1.clicked.connect(self.randomizePool)
+        btn2 = app.getButton(TINY, "see all")
+        btn2.clicked.connect(self.showAll)
+        wgt, _ = Q.makeSeries(Q.HORIZONTAL, btn1, Q.STRETCH, btn2)
+        self.layout.addWidget(wgt)
+
+    def getPools(self):
+        """
+        Get the current master list of VSPs from decred.org.
+        """
+        net = self.app.dcrdata.params
+        def get():
+            try:
+                return StakePool.providers(net)
+            except Exception as e:
+                log.error("error retrieving stake pools: %s" % e)
+                return False
+        self.app.makeThread(get, self.setPools)
+
+    def setPools(self, pools):
+        """
+        Cache the list of stake pools from decred.org, and pick one to display.
+
+        Args:
+            pools (list(object)): The freshly-decoded-from-JSON stake pools.
+        """
+        if not pools:
+            return
+        self.pools = pools
+        tNow = int(time.time())
+        # only save pools updated within the last day
+        self.pools = [p for p in pools if tNow - p["LastUpdated"] < 86400]
+        # sort by performance
+        self.pools.sort(key=self.scorePool, reverse=True)
+        self.randomizePool()
+
+    def randomizePool(self, e=None):
+        """
+        Randomly select a pool from the best performing half, where performance
+        is based purely on voting record, e.g. voted/(voted+missed). The sorting
+        and some initial filtering was already performed in setPools.
+        """
+        count = len(self.pools)//2+1
```

Comment:
This is an simple but pretty dumb way to filter low-performing pools. Consider other options.